packages/core-app-api/src/apis/implementations/StorageApi/WebStorage.ts

```diff
@@ -89,9 +101,19 @@ export class WebStorage implements StorageApi {
   observe$<T extends JsonValue>(
     key: string,
   ): Observable<StorageValueSnapshot<T>> {
+    if (!WebStorage.hasSubscribers) {
+      WebStorage.addStorageEventListener();
+      WebStorage.hasSubscribers = true;
+    }
     return this.observable.filter(({ key: messageKey }) => messageKey === key);
   }
 
+  handleStorageChange(eventKey: StorageEvent['key']) {
```

Comment:
This should still be private (making it go away from the api-report file). `addStorageEventListener` can still access the private method right, since it's inside the class itself? Otherwise do `(webStorage as any). handleStorageChange `

Reply:
> addStorageEventListener can still access the private method right, since it's inside the class itself? 

Yes, you are right. I just left it public since I also defined it in `StorageApi` interface, in case other classes implement it in the future. 

Making it private ...

Reply:
Done https://github.com/backstage/backstage/pull/15040/commits/1dec14de936bf550d5252b34055f5020c500fb29